raw-data/fotmob-leagues/get_fotmob_leagues.R

```diff
@@ -9,28 +8,90 @@ library(tidyr)
 library(janitor)
 library(here)
 
-page <- 'https://www.fotmob.com/' %>% 
+page <- "https://www.fotmob.com/" %>% 
   rvest::read_html()
 
-nested_leagues <- page %>% 
-  rvest::html_elements('script') %>% 
+league_element <- page %>% 
+  rvest::html_elements("script") %>% 
   rvest::html_text(trim = TRUE) %>% 
-  purrr::keep(stringr::str_detect, 'lsLeague') %>% 
+  purrr::keep(stringr::str_detect, "lsLeague")
+
+all_leagues <- league_element %>% 
   stringr::str_extract(
     '(?<=\\"countries\\"[:]).*(?=\\,\\"userSettings\\":null[}]\\,\\"userSettings\\")'
   ) %>% 
   jsonlite::fromJSON() %>% 
-  tidyr::as_tibble()
-
-leagues <- nested_leagues %>%
+  tibble::as_tibble() %>%
   dplyr::rename(
     country = name
   ) %>% 
   tidyr::unnest(leagues) %>% 
   janitor::clean_names()
 
+popular_leagues <- league_element %>% 
+  stringr::str_extract(
+    '(?<=\\"popular\\"[:]).*(?=\\,\\"international\\")'
+  ) %>% 
+  jsonlite::fromJSON() %>% 
+  tibble::as_tibble() %>% 
+  janitor::clean_names()
+
 write.csv(
-  leagues, 
-  here::here('raw-data', 'fotmob-leagues', 'all_leagues.csv'), 
+  all_leagues, 
+  here::here("raw-data", "fotmob-leagues", "all_leagues.csv"), 
   row.names = FALSE
 )
+
+## Manually go to league page -> stats tab -> (See all) under any stat card -> extract season id from the url
+## This is the only way that I could find season ids. This is needed to get the list of season ids for select leagues.
+# nested_leagues$page_url %>% basename() %>% clipr::write_clip()
+season_ids_2022 <- c(
+  "EURO" = 16215,
+  "Champions League" = 16407,
+  "Copa America" = 15148,
+  "Europa League" = 16640,
+  "Premier League" = 16390,
+  "1. Bundesliga" = 16494,
+  "LaLiga" = 16520,
+  "Ligue 1" = 16499,
+  "MLS" = 17185,
+  "Serie A" = 16621
+)
```

Comment:
other leagues could be added here, e.g. Portugal's top league's 2022 season id is 16582, but i've just started with these since it covers what most users would be looking for.

Reply:
Yeah awesome, might be good to add more as time goes on